### ISSUE TITLE
[no-conditional-statement] "allowReturningBranches" should allow branches with `never`-returning functions

### ISSUE THREAD

**Issue by bitjson** [jonaskello/eslint-plugin-functional#99]
As of TypeScript 3.7, [`never`-returning functions now affects the control flow graph](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#better-support-for-never-returning-functions). So now code like below produces an error:
```ts
  if (condition) {
    process.exit();
    return; // Unreachable code detected.ts(7027)
  }
```
However, if the `return` is removed, `no-conditional-statement` with `allowReturningBranches` produces an error:
```ts
  if (condition) {
    process.exit();
  } // Incomplete branch, every branch in a conditional statement must contain a return statement. (eslint) functional/no-conditional-statement
```

**Comment by bitjson** [comment]
On a related note, `throw` statements should also be treated the same way, since TypeScript recognizes that code beyond them is unreachable:
```ts
  if (condition) {
    throw new Error('test');
    return; // Unreachable code detected.ts(7027)
  }
```

**Comment by kristiannotari** [comment]
This would be great to have, both for process.exit and for throw statements
